extensions/smallrye-rest-client/deployment/src/main/java/io/quarkus/smallrye/restclient/deployment/SmallRyeRestClientProcessor.java

```diff
@@ -229,6 +230,18 @@ public void register(RegistrationContext registrationContext) {
         RestClientBuilderImpl.SSL_ENABLED = sslNativeConfig.isEnabled();
     }
 
+    @BuildStep
+    void registerProviders(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            JaxrsProvidersToRegisterBuildItem jaxrsProvidersToRegisterBuildItem) {
+        RestClientBuilderImpl.REGISTER_BUILTIN_PROVIDERS = jaxrsProvidersToRegisterBuildItem.useBuiltIn();
+        RestClientBuilderImpl.PROVIDERS_TO_REGISTER = String.join(",", jaxrsProvidersToRegisterBuildItem.getProviders());
+
+        // register the providers for reflection
+        for (String providerToRegister : jaxrsProvidersToRegisterBuildItem.getProviders()) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, providerToRegister));
```

Comment:
I think the reason why your jar is a bit bigger is that you didn't remove that code: https://github.com/quarkusio/quarkus/pull/1619/files#diff-90fd4c56e0f60bb0cb22626be37bea55R98 . So you add additional reflection rules without removing any code in the end.

Note that you will **need** to support the detected providers in `RestClientBuilderImpl`. Otherwise you will have some issues with providers not being available.

I also think that https://github.com/quarkusio/quarkus/pull/1619/files#diff-90fd4c56e0f60bb0cb22626be37bea55R92 should probably be conditioned to the presence of the JSON binding provider in the list (or the built-in case).

Reply:
Not sure why but the diff links don't show me anything specific, just this page at the top.

`RestClientBuilderImpl` registers the providers with RESTEasy through the properties. I'm not sure what else you're expecting it to do

Reply:
GitHub is flaky today, I suppose that's why it doesn't bring you to the right line.

Trying with another simpler link: here you still register all the providers for reflection: https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-rest-client/deployment/src/main/java/io/quarkus/smallrye/restclient/deployment/SmallRyeRestClientProcessor.java#L97 . Which means that they will all be embedded in the native image.

The whole point of studying the `@Consumes`/`@Produces` annotations it to not do that.

Reply:
Ugh, that's my bad, completely missed that!

I imagine the Abstract JSON-B provider a bit further up should also be removed?

Reply:
I think it should be conditioned on the presence of `org.jboss.resteasy.plugins.providers.jsonb.JsonBindingProvider` in the list.

Reply:
But thinking about it more, it should be moved to the common part and be added if `providersToRegister` contains `org.jboss.resteasy.plugins.providers.jsonb.JsonBindingProvider`.

Reply:
Thinking the same, so will do that